linalg/hypre.cpp

```diff
@@ -216,6 +228,15 @@ HypreParVector& HypreParVector::operator=(const HypreParVector &y)
    return *this;
 }
 
+HypreParVector& HypreParVector::operator=(HypreParVector &&y)
+{
+   // If the argument vector owns its data, then the calling vector will as well
+   WrapHypreParVector(static_cast<hypre_ParVector*>(y), y.own_ParVector);
+   // Either way the argument vector will no longer own its data
+   y.own_ParVector = 0;
```

Comment:
Would it make sense to set `y`'s `hypre_ParVector` member data to `nullptr` (the same way you zero-out the data and size in the `Vector` move assignment operator)?